File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp

class Solution {
public:
    int sumNumbers(TreeNode* root) {
     
       return dfs(root,0);
    }
    int dfs(TreeNode* root,int sum){
        if(root==NULL) return 0;
        sum=sum*10+root->val;
        if(root->left==NULL && root->right==NULL) return sum;
        int a=dfs(root->left,sum);
        int b=dfs(root->right,sum);
        return a+b;
    }
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java

class Solution {
    public int sumNumbers(TreeNode root) {
      return helper(root,0);  
    }
    public int helper(TreeNode root,int sum){
        if(root==null){
            return 0;
         }
         if(root.left==null && root.right==null) return sum*10+root.val;
         sum=root.val+sum*10;
         int a=helper(root.left,sum);
         int b=helper(root.right,sum);
         return a+b;
    }
}